Explorer/Assets/DCL/Notifications/NotificationsMenu/NotificationsMenuController.cs

```diff
@@ -0,0 +1,113 @@
+using CommunicationData.URLHelpers;
+using Cysharp.Threading.Tasks;
+using DCL.Notification.NotificationEntry;
+using DCL.Notification.NotificationsBus;
+using DCL.SidebarBus;
+using DCL.Utilities;
+using DCL.WebRequests;
+using SuperScrollView;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using Utility;
+
+namespace DCL.Notification.NotificationsMenu
+{
+    public class NotificationsMenuController
+    {
+        private const int PIXELS_PER_UNIT = 50;
+
+        private readonly NotificationsMenuView view;
+        private readonly NotificationsRequestController notificationsRequestController;
+        private readonly INotificationsBusController notificationsBusController;
+        private readonly NotificationIconTypes notificationIconTypes;
+        private readonly IWebRequestController webRequestController;
+        private readonly ISidebarBus sidebarBus;
+        private readonly Dictionary<string, Sprite> notificationThumbnailCache = new ();
+        private readonly List<INotification> notifications = new ();
+
+        public NotificationsMenuController(
+            NotificationsMenuView view,
+            NotificationsRequestController notificationsRequestController,
+            INotificationsBusController notificationsBusController,
+            NotificationIconTypes notificationIconTypes,
+            IWebRequestController webRequestController,
+            ISidebarBus sidebarBus)
+        {
+            this.view = view;
+            this.notificationsRequestController = notificationsRequestController;
+            this.notificationsBusController = notificationsBusController;
+            this.notificationIconTypes = notificationIconTypes;
+            this.webRequestController = webRequestController;
+            this.sidebarBus = sidebarBus;
+            this.view.LoopList.InitListView(0, OnGetItemByIndex);
+            this.view.CloseButton.onClick.AddListener(ClosePanel);
+
+            InitialNotificationRequest().Forget();
+        }
+
+        private void ClosePanel()
+        {
+            sidebarBus.UnblockSidebar();
+            view.gameObject.SetActive(false);
+        }
+
+        public void ToggleNotificationsPanel(bool forceClose)
+        {
+            view.gameObject.SetActive(!forceClose && !view.gameObject.activeSelf);
+        }
+
+        private async UniTaskVoid InitialNotificationRequest()
+        {
+            List<INotification> requestNotifications = await notificationsRequestController.RequestNotifications();
+
+            foreach (INotification requestNotification in requestNotifications)
+                notifications.Add(requestNotification);
+
+            view.LoopList.SetListItemCount(notifications.Count, false);
+        }
+
+        private LoopListViewItem2 OnGetItemByIndex(LoopListView2 loopListView, int index)
```

Comment:
Maybe we can rename this `LoopListViewItem2` to another more descriptive name. We shouldn't have this kind of names ended in numbers in the types.

Reply:
The name is coming from a plugin we are using that defines the class like that, we cannot change it :(